salt/states/ssh_auth.py

```diff
@@ -324,16 +308,11 @@ def present(
                     fingerprint_hash_type=fingerprint_hash_type,
                 )
             else:
-                # Split keyline to get key and comment
-                keyline = keyline.split(" ")
-                key_type = keyline[0]
-                key_value = keyline[1]
-                key_comment = keyline[2] if len(keyline) > 2 else ""
                 data = __salt__["ssh.set_auth_key"](
```

Comment:
The module code may also need updating, if it makes the same assumptions about the file format.